addons/product_expiry/views/stock_quant_views.xml

```diff
@@ -23,8 +23,7 @@
         <field name="arch" type="xml">
             <xpath expr="//field[@name='lot_id']" position="after">
                 <field name="use_expiration_date" invisible="1"/>
-                <field name="expiration_date" optional="hide" invisible="context.get('hide_removal_date')"
-                    attrs="{'readonly': ['|', ('tracking', '=', 'none'), ('use_expiration_date', '=', False)]}"/>
+                <field name="expiration_date" optional="hide" invisible="context.get('hide_removal_date')"/>
```

Comment:
Same change should be done in `view_stock_quant_tree_inventory_editable` too I guess :p

Also, what about `removal_date` ? The first bug doesn't happen for this field, but the second one (the *nuisance*) happens also for this field.

(Anyway, thanks for fixing my crap and sorry :grimacing: )

Reply:
Good catch. Removed it from that other view.

I think removal_date is ok since it's in a readonly view by default + it doesn't affect other field values. It's also been there a long time/ no one has complained about it :thinking: